OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -121,15 +125,29 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			else
 				speed = info.Speed[0];
 
-			// Both the head and tail start at the source actor, but initially only the head is travelling.
+			// Both the head and tail start at the source actor, but initially only the head is traveling.
 			headPos = args.Source;
 			tailPos = headPos;
 
 			target = args.PassiveTarget;
 			if (info.Inaccuracy.Length > 0)
 			{
 				var inaccuracy = Util.ApplyPercentageModifiers(info.Inaccuracy.Length, args.InaccuracyModifiers);
-				var maxOffset = inaccuracy * (target - headPos).Length / args.Weapon.Range.Length;
+
+				var maxOffset = 0;
+				switch (info.InaccuracyType)
```

Comment:
Please pull this out into a helper method instead of duplicating it everywhere!

Reply:
This was my original intention as well, but adding a base class and all the nice things that come with it makes this a lot harder to review, so I opted for doing it in a follow-up PR.